compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1402,21 +1402,21 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
     /// Given a function definition like:
     ///
     /// ```rust
+    /// use std::fmt::Debug;
     /// fn test<'a, T: Debug>(x: &'a T) -> impl Debug + 'a {
     ///     x
     /// }
     /// ```
     ///
-    /// we will create a TAIT definition in the HIR like
-    ///
-    /// ```
-    /// type TestReturn<'a, T, 'x> = impl Debug + 'x
-    /// ```
-    ///
-    /// and return a type like `TestReturn<'static, T, 'a>`, so that the function looks like:
+    /// we will create a TAIT definition in the HIR like and return a type like `TestReturn<'a, T>`,
+    /// so that the function looks like:
```

Comment:
```suggestion
    /// we will create a TAIT definition in the HIR and change the return type so that the function looks like:
```
The `TestReturn` doesn't really make sense without introducing it first. Although I prefer the way it looked previously, but adding a `use` and `#![feature]` would maybe make it too verbose.